lib/inventory/controller.go

```diff
@@ -371,6 +372,11 @@ func (c *Controller) handleControlStream(handle *upstreamHandle) {
 				}
 			case proto.UpstreamInventoryPong:
 				c.handlePong(handle, m)
+			case proto.UpstreamInventoryDelete:
+				if err := c.handleTermination(handle, m); err != nil {
+					handle.CloseWithError(err)
+					return
+				}
```

Comment:
If this message is going to effectively terminate the stream, consider naming it `UpstreamInventoryGoodbye` and having the deletion behavior be controlled by a field.  That way we can easily add any other future "on exit" stuff to this message.